src/components/Connection/Desktop/ImportAccount.vue

```diff
@@ -0,0 +1,291 @@
+        return this.t('desktop.heading.accountDetailsTitle');
+      default:
+        return '';
+    }
+  }
+
+  get disabledNextStep(): boolean {
+    return this.mnemonicPhrase.length === 0;
+  }
+
+  get disabledImportStep(): boolean {
+    return !(!!this.accountName && !!this.accountPassword);
+  }
+
+  get inputType(): string {
+    return this.hiddenInput ? 'password' : 'text';
+  }
+
+  get iconPasswordStyle(): string {
+    return this.hiddenInput ? 'basic-eye-no-24' : 'basic-filterlist-24';
+  }
+
+  toggleVisibility(): void {
+    this.hiddenInput = !this.hiddenInput;
+  }
+
+  handleMnemonicInput(char) {
+    const letter = char.replace('.', '').replace('  ', ' ');
+
+    if (/^[a-z ]+$/.test(letter)) this.mnemonicPhrase = letter;
+  }
+
+  nextStep(): void {
+    if (this.mnemonicPhrase.trim().split(' ').length !== this.PHRASE_LENGTH) {
+      this.$notify({
+        message: this.t('desktop.errorMessages.mnemonicLength', { number: this.PHRASE_LENGTH }),
+        type: 'error',
+        title: '',
+      });
+      this.mnemonicPhrase = '';
+      return;
+    }
+
+    if (!mnemonicValidate(this.mnemonicPhrase)) {
+      this.$notify({
+        message: this.t('desktop.errorMessages.mnemonic'),
+        type: 'error',
+        title: '',
+      });
+      this.mnemonicPhrase = '';
+      return;
+    }
+
+    this.$emit('stepChange', LoginStep.ImportCredentials);
+  }
+
+  importJson(): void {
+    this.fileInput.click();
+  }
+
+  async handleUploadJson(e): Promise<void> {
+    try {
+      const jsonFile = e.target.files[0];
+      if (!jsonFile) {
+        return;
+      }
+
+      const parsedJson = await parseJson(jsonFile);
+
+      try {
+        const { address, encoded, encoding } = parsedJson;
+        const name = parsedJson.meta.name as string;
+
+        if (!address && !encoded && !encoding && !name) {
+          throw new Error('JSON file does not contain required fields.');
+        }
+
+        this.accountName = name;
+        this.readonlyAccountName = true;
+        this.json = parsedJson;
+        this.$emit('stepChange', LoginStep.ImportCredentials);
+      } catch {
+        this.$notify({
+          message: this.t('desktop.errorMessages.jsonFields'),
+          type: 'error',
+          title: '',
+        });
+
+        return;
+      }
+    } catch (err) {
+      this.$notify({
+        message: this.t('unknownErrorText'),
+        type: 'error',
+        title: '',
+      });
+    }
+  }
+
+  async importAccount(): Promise<void> {
+    if (this.json) {
+      this.handleJsonInput(this.json);
+    } else {
+      this.handleCredentialsInput();
+    }
+  }
+
+  async handleJsonInput(json: KeyringPair$Json): Promise<void> {
+    try {
+      api.restoreFromJson(json, this.accountPassword);
+      await this.getPolkadotJsAccounts();
+      this.$emit('stepChange', LoginStep.AccountList);
+    } catch (error) {
+      if (error.message === 'Unable to decode using the supplied passphrase') {
```

Comment:
if error.message will be changed - this logic will be broken

Reply:
so far it is the only known way to infer from problem, 
in else-statement we have common error handling